app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt

```diff
@@ -141,29 +145,42 @@ class ConversationViewModel @Inject constructor(
                 attachmentBundle?.run {
                     when (attachmentType) {
                         AttachmentType.IMAGE -> {
-                            val (imgWidth, imgHeight) = extractImageParams(attachmentBundle.rawContent)
-                            sendImageMessage(
-                                conversationId = conversationId,
-                                imageRawData = attachmentBundle.rawContent,
-                                imageName = attachmentBundle.fileName,
-                                imgWidth = imgWidth,
-                                imgHeight = imgHeight
-                            )
+                            if (rawContent.size > IMAGE_SIZE_LIMIT_BYTES) onError(ErrorMaxImageSize)
+                            else {
+                                val (imgWidth, imgHeight) = extractImageParams(attachmentBundle.rawContent)
+                                val result = sendImageMessage(
+                                    conversationId = conversationId,
+                                    imageRawData = attachmentBundle.rawContent,
+                                    imageName = attachmentBundle.fileName,
+                                    imgWidth = imgWidth,
+                                    imgHeight = imgHeight
+                                )
+                                if (result is SendImageMessageResult.Failure) {
+                                    onError(ErrorSendingSize)
```

Comment:
This is similar as the comment about this class naming, we should use something more generic like `ErrorSendingAsset`